### PR TITLE
JP-841 update master_background docs

### PR DESCRIPTION
**Pull request by hbushouse**
Add some more details to the `master_background` step docs regarding things like how the source type designation comes into play, the type of extractions from `extract_1d`, etc.

### CODE REVIEW

**Comment by hbushouse** [comment]
Fixes #3774

**Comment by hbushouse** [comment]
Reviewers: hold off on review for a bit. I just realized I forgot to add a few sentences about how NIRSpec MOS is handled right now. Let me add that and then it'll be ready for review.

**Comment by hbushouse** [comment]
OK, everything's now ready for review. Have at it.

File: docs/jwst/master_background/description.rst [hunk 1]
```diff
@@ -76,49 +100,147 @@ exposures looks like the following::
...
   }
 
-As you can see from the above ASN list, the same two exposures are defined as
+As you can see, the same two exposures are defined as
 being both "science" and "background" members, because they both contain the target
 of interest and a region of background. The "science" members, which are the
 :ref:`cal <cal>` products created by the :ref:`calwebb_spec2 <calwebb_spec2>`
 pipeline, are the data files that will have the master background subtraction
-applied, while the "background" members are :ref:`x1d <x1d>` 1-D spectral
+applied, while the "background" members are the :ref:`x1d <x1d>` 1-D spectral
```
Comment (jdavies-st) [inline docs/jwst/master_background/description.rst]: "extracted from the background region"

File: docs/jwst/master_background/description.rst [hunk 2]
```diff
@@ -76,49 +100,147 @@ exposures looks like the following::
...
+"BKGDTARG" set to True in the header. During :ref:`calwebb_spec2 <calwebb_spec2>` processing,
+the :ref:`srctype <srctype_step>` step recognizes these and sets their source type to
+"EXTENDED", because all dedicated background exposures are to be processed as extended sources.
+
+This in turn causes the :ref:`extract_1d <extract_1d_step>` step at
+the end of :ref:`calwebb_spec2 <calwebb_spec2>` to extract a spectrum in extended source mode,
+which uses the entire field-of-view (whether it be a slit image or an IFU cube) as the
+extraction region. The extracted spectral data are stored in the "FLUX" and "SURF_BRIGHT"
+columns of the resulting :ref:`x1d <x1d>` product, with the "BACKGROUND" column left blank.
+The master_background step recognizes when it's working with a background exposure
```
Comment (jdavies-st) [inline docs/jwst/master_background/description.rst]: "with a background exposure, an extended source, and in that case"

File: docs/jwst/master_background/description.rst [hunk 3]
```diff
@@ -76,49 +100,147 @@ exposures looks like the following::
...
+members, and two exposures of the background target, labeled as "background"
+members. As before, the science members use :ref:`cal <cal>` products as input
+and the background members use :ref:`x1d <x1d>` products as input.
+The master background step will first combine the data from the two background
+members into a master background spectrum and then subtract it from each of the
+two science exposures.
+
+NIRSpec MOS with Background Slits
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+NIRSpec MOS exposures that have one or more slits defined as background only require
```
Comment (jdavies-st) [inline docs/jwst/master_background/description.rst]: ~~only~~

File: docs/jwst/master_background/description.rst [hunk 4]
```diff
@@ -76,49 +100,147 @@ exposures looks like the following::
...
+unique processing. The background slits take the place of both nodding and off-target
```
Comment (jdavies-st) [inline docs/jwst/master_background/description.rst]: nodding -> nodded

File: docs/jwst/master_background/description.rst [hunk 5]
```diff
@@ -76,49 +100,147 @@ exposures looks like the following::
...
+only workable option for NIRSpec MOS data at this time is to employ the
+user-supplied background spectrum option, which is then subtracted from every
+slit instance in a MOS exposure.
 
-Creating the 1-D master background spectrum
+Creating the 1-D Master Background Spectrum
 -------------------------------------------
 The 1-D master background spectrum is created by combining data contained in the
-:ref:`x1d <x1d>` products listed in the input ASN as being "exptype: background" members.
+:ref:`x1d <x1d>` products listed in the input ASN as "exptype: background" members.
```
Comment (jdavies-st) [inline docs/jwst/master_background/description.rst]: `"exptype": "background"`

File: docs/jwst/master_background/description.rst [hunk 6]
```diff
@@ -11,38 +11,62 @@ versus wavelength - is projected into the
...
+as input to the :ref:`calwebb_spec3 <calwebb_spec3>` pipeline, which defines a stage 3 combined product
 and its input members. The list of input members includes both "science" and "background"
 exposure types. The master background subtraction step uses the input members designated
-with "exptype: background" as the data from which to create the master background spectrum.
-These should be :ref:`x1d <x1d>` products created for individual exposures at the end of
-the :ref:`calwebb_spec2 <calwebb_spec2>` pipeline.
-The master background will be subtracted from all input members designated as
-"exptype: science" in the ASN, resulting in a new version of each such input. These inputs
-should be :ref:`cal <cal>` products created for individual exposures by the
+with "exptype: background" to create the master background spectrum.
```
Comment (philhodge) [inline docs/jwst/master_background/description.rst]: Shouldn't this be "exptype": "background"?

File: docs/jwst/master_background/description.rst [hunk 7]
```diff
@@ -11,38 +11,62 @@ versus wavelength - is projected into the
...
+for measuring background. Exposures of this type are identified by the pipeline based on their
+"PATTTYPE" (primary dither pattern type) keyword value. The value will either contain the
+substring "NOD" somewhere within the name (e.g. "2-POINT-NOD" or "ALONG-SLIT-NOD"), or will
+be set to "POINT-SOURCE" (for MIRI MRS).  The :ref:`calwebb_spec2 <calwebb_spec2>`
+:ref:`srctype <srctype_step>` step recognizes these PATTTYPE values and sets the
+source type to "POINT".
+
+This in turn causes the :ref:`extract_1d <extract_1d_step>` step at
+the end of :ref:`calwebb_spec2 <calwebb_spec2>` to extract spectra for both source and
+background regions. For IFU exposures the background region is typically an annulus that is
```
Comment (philhodge) [inline docs/jwst/master_background/description.rst]: For IFU data, it is true that a background annulus will be subtracted by default.  For non-IFU data, however, a background will only be subtracted in extract_1d if one or more background regions are specified in the reference file.  There may be some extract_1d reference files in CRDS that do define background regions, but many do not.

**Pull request merged**